hare/preroundtracker_test.go

```diff
@@ -23,7 +23,30 @@ const (
 )
 
 func getProposalID(i int) types.ProposalID {
-	return types.GenLayerProposal(types.NewLayerID(1), nil).ID()
+	return genLayerProposal(types.NewLayerID(1), nil).ID()
+}
+
+func genLayerProposal(layerID types.LayerID, txs []types.TransactionID) *types.Proposal {
```

Comment:
i don't get why removing this from common/testutils? i can support moving the testutils to a different pkg maybe.

Reply:
We already have a `common/utils` package that I cannot move this to as otherwise I introduce a cyclic dependency. Creating a new `common/testutils` package for this creates another point of friction that will cause issues of cyclic dependencies in the future.

This is btw. another good example of why I think this "do not repeat yourself" with helpers is well intentioned but usually more trouble than what it's worth: This function is called in 3 places.

- `hare/preroundtracker_test.go` function `getProposalID` which can just return a random ID instead.
- `hare/hare_test.go` function `randomProposal` which overwrites half the fields in the proposal and creates a new signature, at that point why call `genLayerProposal` in the first place?
- `hare/hare_rounds_test.go`; the only legitimate use of this helper
- `hare/flows_test.go`; which overwrites the epoch data of the generated proposal rendering its signature invalid

Especially in tests I feel like we are re-using too much code between tests leading to weird and hard to debug errors and race conditions. Imo we should stop doing that and we would have more self contained and thereby stable tests.